### PR TITLE
Add Ammonite Ivy completions for Scala 3

### PR DESCRIPTION
**Pull request by jkciesluk**
Adds ammonite ivy imports completions for Scala 3.
Also refactors these completions for Scala 2 to use `CoursierComplete`.

Fixes https://github.com/scalameta/metals/issues/4551

### CODE REVIEW

File: mtags/src/main/scala-3/scala/meta/internal/pc/completions/CompletionValue.scala [hunk 1]
```diff
@@ -154,7 +154,7 @@ object CompletionValue:
     override def completionItemKind(using Context): CompletionItemKind =
       CompletionItemKind.File
 
-  case class ScalaCLiImport(
+  case class IvyImport(
```
Comment (jkciesluk) [inline mtags/src/main/scala-3/scala/meta/internal/pc/completions/CompletionValue.scala:157]: Can we have one `CompletionValue` for both ScalaCli and Ammonite dependency imports?

Comment (tgodzik) [reply mtags/src/main/scala-3/scala/meta/internal/pc/completions/CompletionValue.scala:157]: Sure, it's probably better, we could even hoave just one for sbt as well.

Comment (tgodzik) [reply mtags/src/main/scala-3/scala/meta/internal/pc/completions/CompletionValue.scala:157]: ach right :sweat_smile:

**Review by tgodzik — APPROVED** [review]
LGTM!

**Pull request merged**